compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -227,3 +229,13 @@ pub fn lower_ty<'tcx>(tcx: TyCtxt<'tcx>, hir_ty: &hir::Ty<'tcx>) -> Ty<'tcx> {
     let env_def_id = tcx.hir().get_parent_item(hir_ty.hir_id);
     collect::ItemCtxt::new(tcx, env_def_id.def_id).lower_ty(hir_ty)
 }
+
+/// This is for rustdoc and clippy.
+pub fn lower_const_arg<'tcx>(
```

Comment:
The whole story of consts in rustdoc is a mess (see #83035). The TLDR is that we use both `hir::Const` and `ty::Const` and we basically need to use both. It's an open question whether/in which cases rustdoc should show the original const expression (`hir`) or the evaluated version (`ty`). There's advantages to both. So in the middle of processing some `hir` nodes, rustdoc may need to lower a hir const to a ty const. I don't think we'll be able to get rid of that anytime soon.

You're right that clippy doesn't need it though; I think it did when I originally wrote this code but was since modified to no longer need it.